tests/quantization/test_quantized_module.py

```diff
@@ -351,6 +351,110 @@ def test_quantized_module_rounding_fhe(model_class, input_shape, default_configu
     # TODO: https://github.com/zama-ai/concrete-ml-internal/issues/3800
 
 
+@pytest.mark.parametrize("model_class, input_shape", [pytest.param(FC, (100, 32 * 32 * 3))])
+def test_statuses(model_class, input_shape, default_configuration):
+    """Check that giving statuses work properly."""
+
+    torch_fc_model = model_class(activation_function=nn.ReLU)
+    torch_fc_model.eval()
+
+    # Create random input
+    numpy_input = numpy.random.uniform(size=input_shape)
+    torch_input = torch.from_numpy(numpy_input).float()
+    assert isinstance(torch_input, torch.Tensor)
+    assert isinstance(numpy_input, numpy.ndarray)
+
+    # Compile with rounding activated
+    with pytest.raises(ValueError):
+        # Empty status
+        compile_torch_model(
+            torch_fc_model,
+            torch_input,
+            False,
+            default_configuration,
+            n_bits=2,
+            p_error=0.01,
+            rounding_threshold_bits=6,
+            statuses={},
+        )
+
+    # Wrong name status
+    with pytest.raises(ValueError):
+        compile_torch_model(
+            torch_fc_model,
+            torch_input,
+            False,
+            default_configuration,
+            n_bits=2,
+            p_error=0.01,
+            rounding_threshold_bits=6,
+            statuses={"x": "encrypted"},
+        )
+
+    # Wrong encryption status
+    with pytest.raises(ValueError):
+        compile_torch_model(
+            torch_fc_model,
+            torch_input,
+            False,
+            default_configuration,
+            n_bits=2,
+            p_error=0.01,
+            rounding_threshold_bits=6,
+            statuses={"_onnx__Gemm_0": "random"},
+        )
+
+    # Additional argument (error from Concrete Python)
+    with pytest.raises(ValueError):
+        compile_torch_model(
+            torch_fc_model,
+            torch_input,
+            False,
+            default_configuration,
+            n_bits=2,
+            p_error=0.01,
+            rounding_threshold_bits=6,
+            statuses={"_onnx__Gemm_0": "encrypted", "test": "encrypted"},
+        )
+
+    # No encrypted value
+    with pytest.raises(RuntimeError):
```

Comment:
this is a CP restriction right ? is there a clear (🙂) message associated to it ? If not, maybe worth to add our own one, else maybe add a `match=` here as well

Reply:
If there is no encrypted value CP can't compile indeed.

Their error message is quite explicit, I'll add a `match=""`